src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertyAccessorSymbolReferenceFinder.cs

```diff
@@ -19,29 +18,86 @@ protected override bool CanFind(IMethodSymbol symbol)
         {
-            return FindDocumentsAsync(project, documents, cancellationToken, symbol.Name);
+            var namesToIncludeInSearch =
+                options.IncludeImplicitAccessorUsages && symbol.AssociatedSymbol is IPropertySymbol property
+                    ? new[] { symbol.Name, property.Name }
+                    : new[] { symbol.Name };
+
+            return FindDocumentsAsync(
+                project,
+                documents,
+                documentIndex => namesToIncludeInSearch.Any(documentIndex.ProbablyContainsIdentifier),
+                cancellationToken);
         }
 
-        protected override Task<ImmutableArray<ReferenceLocation>> FindReferencesInDocumentAsync(IMethodSymbol symbol, Document document, SemanticModel semanticModel, CancellationToken cancellationToken)
+        protected override async Task<ImmutableArray<ReferenceLocation>> FindReferencesInDocumentAsync(IMethodSymbol symbol, Document document, SemanticModel semanticModel, SymbolFinderOptions options, CancellationToken cancellationToken)
         {
-            return FindReferencesInDocumentUsingSymbolNameAsync(symbol, document, semanticModel, cancellationToken);
+            var explicitReferences = await FindReferencesInDocumentUsingSymbolNameAsync(symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
+
+            var property = symbol.AssociatedSymbol as IPropertySymbol;
+            if (property == null || !options.IncludeImplicitAccessorUsages)
+            {
+                return explicitReferences;
+            }
+
+            // Picks up whatever an IPropertySymbol search would find, e.g. foreach and element access.
+            // Each of these is a candidate implicit reference.
+            var propertyReferences = await ReferenceFinders.Property.FindReferencesInDocumentAsync(
+                SymbolAndProjectId.Create(property, document.Project.Id),
+                document,
+                semanticModel,
+                options,
+                cancellationToken).ConfigureAwait(false);
+
+            var results = ImmutableArray.CreateBuilder<ReferenceLocation>();
+            results.AddRange(explicitReferences);
+
+            var isSetterSearch = symbol == property.SetMethod;
+
+            foreach (var propertyReference in propertyReferences)
+            {
+                // IsWrittenTo is a hack since it can't distinguish between:
+                //  - nameof(Property) (no accessor—mistakenly included in getter search)
+                //  - _ = Property (getter)
+                //  - Property++ (getter *and* setter—mistakenly excluded from getter search)
+                // And IsWrittenTo will be useless for event accessors.
+
+                // We need the proposed SemanticModel.GetAccessors at
+                // https://github.com/dotnet/roslyn/issues/25057#issuecomment-368659777
+                if (propertyReference.IsWrittenTo == isSetterSearch)
```

Comment:
Going to need help with this, differentiating `nameof(P)`, `_ = P`, `P = default`, and `P++`.
Since this has to be language agnostic I'm not sure how massive an undertaking it might be.

Or maybe we live with the hack and fix it after https://github.com/dotnet/roslyn/issues/25057?